scripts/bumpedTemplateVersion.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+# What this does:
+# 1. Take the version of react, use the MAJOR.MINOR
+#     - /nightly/ -> MAJOR.MINOR.PATCH-nightly-YYYYNN-HASH
+#     - /-rc.N/   -> MAJOR.MINOR.PATCH-rc.N-HASH
+#     - ELSE: continue
+# 2. Look for the latest version of the template with that MAJOR.MINOR
+#   - No version -> then set to MAJOR.MINOR.0
+#   - MAJOR.MINOR.PATCH -> MAJOR.MINOR.PATCH+1
+
+if [[ $# != 1 ]]; then
+  echo "USAGE: bumpTemplateVersion.sh <react native version>"
+  exit 1
+fi
+
+
+if [[ $1 =~ -rc\.[0-9]+ ]]; then
+  echo "Release candidate"
+  # Append a sha, so if we need to re-release for a RC we don't conflict
+  echo $1-$(git rev-parse --short HEAD)
```

Comment:
how can the CLI find the latest RC if we attach a sha at the end of the version?

Reply:
IIUC SHA is attached to template's version but React Native version is stored under npm's metadata for each template's version published. CLI after https://github.com/react-native-community/cli/pull/2475 lands will get template's versions and will sort by matching npm's registry metadata (which stores React Native version) and will get the latest one if there are >= 2 matching versions, that's how CLI is able to find version without knowing SHA.

Reply:
That's exactly right @szymonrybczak!